include/navigation/insilico_rotater.h

```diff
@@ -49,31 +48,31 @@ struct InSilicoRotater
             BOB_ASSERT(image.isContinuous());
             BOB_ASSERT(beginRoll < endRoll);
             BOB_ASSERT((distance(endRoll, beginRoll) % scanStep) == 0);
-
-            for (auto &image : m_ScratchImages.elements()) {
-                image.create(unwrapRes, CV_8UC1);
-            }
-            if (!maskImage.empty()) {
-                for (auto &mask : m_ScratchMaskImages.elements()) {
-                    mask.create(unwrapRes, maskImage.type());
-                }
-            }
-       }
+      }
 
         template<class Func>
         void rotate(Func func) const
         {
-            #pragma omp parallel for
-            for (auto i = m_BeginRoll; i < m_EndRoll; i += m_ScanStep) {
-                auto &scratchImage = m_ScratchImages.current();
-                auto &scratchMask = m_ScratchMaskImages.current();
-                const auto index = toIndex(i);
-                rollImage(m_Image, scratchImage, index);
-                if (!scratchMask.empty()) {
-                    rollImage(m_MaskImage, scratchMask, index);
+            #pragma omp parallel
+            {
+                static cv::Mat scratchImage, scratchMask;
+                #pragma omp threadprivate(scratchImage, scratchMask)
+
+                scratchImage.create(m_Image.size(), CV_8UC1);
```

Comment:
Isn't the whole purpose of the scratch images that they **don't** have to get re-allocated every time? This should be doable with function-scope statics but you'd have to replace the create calls with:
```
static cv::Mat scratchImage(m_Image.size(), CV_8UC1);
static cv::Mat scratchMask(m_MaskImage.size(), CV_8UC1);
```

Reply:
Actually, as ``create`` is lazy, I think this is fine but it's probably worth putting a comment to that effect.

Reply:
Yep. Plus this way if the size of ``m_Image`` changes between calls then ``scratchImage`` will still be the right size.

Reply:
I've stuck a comment in now